exo/main.py

```diff
@@ -74,6 +75,7 @@
   for chatgpt_api_endpoint in chatgpt_api_endpoints:
     print(f" - {terminal_link(chatgpt_api_endpoint)}")
 
+device_capabilities = device_capabilities(inference_engine)
```

Comment:
I'd prefer a design where `device_capabilities` is asynchronous and called from the discovery module. That way we don't block exo from getting started and the discovery module can fetch the device capabilities when it needs to. Also, if it fails for whatever reason it can be retried. For now lets cache the device capabilities in the discovery module forever but eventually we'd want to "refresh" it by running the benchmark again

Reply:
I think I've now got it cached correctly in the discovery modules (this is my first time working with the async library so I'm not 100% sure)